src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -148,6 +148,7 @@ public class LockFileBuilder
                 }
             }
 
+            ValidateLockFileLibrary(lockFile);
             var libraries = lockFile.Libraries.ToDictionary(lib => Tuple.Create(lib.Name, lib.Version));
```

Comment:
I'm sorry I missed this earlier, do you think it'd be possible to just have the method above return the dictionary instead? That way we have *no* wasted allocations whatsoever, we just create an extra dictionary.

Reply:
Great point, I refactored the code. I  should have thought about it before. Please check now.